Plutarch/Show.hs

```diff
@@ -58,9 +60,19 @@ class PShow t where
   default pshow' :: (PGeneric t, PlutusType t, All2 PShow (PCode t)) => Bool -> Term s t -> Term s PString
   pshow' wrap x = gpshow wrap # x
 
--- | Return the string representation of a Plutarch value
+{- | Return the string representation of a Plutarch value.
+
+= Important note
+
+Use this in preference to @pshow'@ whenever possible. We use some additional
+logic to avoid compiling (potentially very expensive) 'PShow' code into
+traces when tracing is disabled in this function, which will not be used if
+you write code using @pshow'@ instead.
+-}
 pshow :: PShow a => Term s a -> Term s PString
-pshow = pshow' False
+pshow t = pgetConfig $ \case
+  NoTracing -> pconstant ""
+  _ -> pshow' False t
```

Comment:
I don't think this is a correct solution. This will break other valid use cases for pshow. 

Rather, I think we should try to fix it on trace functions to correctly prune branches on certain config

Reply:
Yeah, upon closer inspection I agree - just because tracing isn't on doesn't mean we want to block all `PShow` use. However, the kind of pruning you describe seems rather hard to do.

Reply:
Why would you ever want to show on-chain? What are the valid use cases?